vendor/bat-native-confirmations/src/bat/confirmations/internal/redeem_token.cc

```diff
@@ -255,13 +233,18 @@ void RedeemToken::OnFetchPaymentToken(
   }
 
   if (response_status_code == net::HTTP_NOT_FOUND) {
+    BLOG(WARNING) << "Confirmation not found";
+
     if (!Verify(confirmation)) {
       BLOG(ERROR) << "Failed to verify confirmation";
       OnRedeem(FAILED, confirmation, false);
       return;
     }
 
-    Redeem(confirmation.creative_instance_id, confirmation.type);
+    ConfirmationInfo new_confirmation = confirmation;
```

Comment:
I think we should do something similar to this before line 270 too. Since if the reply is anything other than OK it’s not safe to assume confirmation is created

Reply:
@amirsaber The current logic will retry for anything other than HTTP_OK and if the server then returns a response of HTTP_NOT_FOUND we recreate the confirmation